tasks/queue-granules/src/index.ts

```diff
@@ -0,0 +1,224 @@
+    ({ collectionName, collectionVersion }) => constructCollectionId(
+      collectionName,
+      collectionVersion
+    )
+  );
+  const parentExecutionArn = buildExecutionArn(
+    get(event, 'cumulus_config.state_machine')!,
+    get(event, 'cumulus_config.execution_name')!
+  )!;
+  const pMapConcurrency = get(event, 'config.concurrency', 3);
+
+  const messageTemplate = await getJsonS3Object(
+    event.config.internalBucket,
+    templateKey(event.config.stackName)
+  );
+  const { arn: granuleIngestWorkflowArn } = await getJsonS3Object(
+    event.config.internalBucket,
+    getWorkflowFileKey(event.config.stackName, event.config.granuleIngestWorkflow)
+  );
+
+  const executionArns = await pMap(
+    createIterable(granules, event.config.preferredQueueBatchSize),
+    async ({ provider, collectionId, chunks }) => {
+      const { name: collectionName, version: collectionVersion } = deconstructCollectionId(
+        collectionId
+      );
+      const [collection, normalizedProvider] = await Promise.all([
+        memoizedFetchCollection({
+          prefix: event.config.stackName,
+          collectionName,
+          collectionVersion,
+        }),
+        memoizedFetchProvider(event, provider),
+      ]);
+
+      return await pMap(
+        chunks,
+        async (granuleBatchIn) => {
+          const granuleBatch = updateGranuleBatchCreatedAt(granuleBatchIn, Date.now());
+          await pMap(
+            granuleBatch,
+            (queuedGranule) => {
+              const { granuleId, updatedAt, createdAt } = queuedGranule
+
+              if (updatedAt && (!Number.isInteger(updatedAt) || updatedAt < 0)) {
+                throw new Error(`Invalid collection information provided for granule with granuleId: ${granuleId}, please check task input to make sure collection information is provided`)
+              }
+              return granulesApi.updateGranule({
+                prefix: event.config.stackName,
+                collectionId,
+                granuleId,
+                body: {
+                  collectionId,
+                  granuleId,
+                  status: 'queued',
+                  updatedAt: updatedAt!,
+                  createdAt: createdAt!,
+                }
+              })
+            },
+            { concurrency: pMapConcurrency }
+          )
+
+          return await enqueueGranuleIngestMessage({
+            messageTemplate,
+            workflow: {
+              name: event.config.granuleIngestWorkflow,
+              arn: granuleIngestWorkflowArn,
+            },
+            granules: granuleBatch,
+            queueUrl: event.config.queueUrl,
+            provider: normalizedProvider,
+            collection,
+            pdr: event.input.pdr,
+            parentExecutionArn,
+            executionNamePrefix: event.config.executionNamePrefix,
+            additionalCustomMeta: event.config.childWorkflowMeta,
+          });
+        },
+        { concurrency: pMapConcurrency }
```

Comment:
Same suggestion as above:

I'd suggest also setting the `pMap` option `stopOnError` to `false`, because I don't believe there's any reason to stop on the first error in the array being mapped. There's no dependency between items in the array, so subsequent items might succeed even after earlier ones fail. By setting `stopOnError` to `false`, you allow as many items to succeed as possible. Otherwise, you fail to process later items in the array as soon as the first error is encountered.

See https://github.com/sindresorhus/p-map#stoponerror

Reply:
https://github.com/nasa/cumulus/pull/3441/commits/b324123d44d80d2bebd8b48903688e4b0bb9b41a#diff-5a524644461453261cb8eed4a96282c50fcd293225634ffdd3ea700eae8d13a8R197